python/ray/data/dataset.py

```diff
@@ -182,20 +182,25 @@ class Dataset:
     via the ``write_*()`` APIs.
```

Comment:
For 
> Datasets are distributed pipelines that produce ``ObjectRef[Block]`` outputs,
    where each block holds data in Arrow format, representing a shard of the overall
    data collection. The block also determines the unit of parallelism.

Could you reference "Data internals"? Also, it's not entirely true that blocks are represented as Arrow tables. They can also be represented with pandas DataFrames.